crypto/s2n_hmac.c

```diff
@@ -298,7 +298,8 @@ int s2n_hmac_digest_two_compression_rounds(struct s2n_hmac_state *state, void *o
      * 17 bytes if the block size is 128.
      */
     const uint8_t space_left = (state->hash_block_size == 128) ? 17 : 9;
-    if (state->currently_in_hash_block > (state->hash_block_size - space_left)) {
+    ENSURE_POSIX(state->currently_in_hash_block <= INT32_MAX, S2N_ERR_INTEGER_OVERFLOW);
+    if ((int32_t)state->currently_in_hash_block > (state->hash_block_size - space_left)) {
```

Comment:
Rather than checking and downcasting the integer size, could we upcast to a int64 instead?

Reply:
Of course!! Thanks, implemented as suggested.